notebooks/ch12/linear_case_study.md

```diff
@@ -0,0 +1,1205 @@
+
+
+```python
+# HIDDEN
+model = LinearRegression()
+model.fit(X_train[['Girth', 'Length']], X_train['Weight'])
+predictions = model.predict(X_test[['Girth', 'Length']])
+print("MSE:", mse_test_set(predictions))
+```
+
+    MSE: 9680.90242337725
+    
+
+Wow! Looks like our MSE went down from around 13000 with just `Girth` alone to 10000 with `Girth` and `Length`. Adding more variables seems to improve our model.
+
+We can also use categorical variables in our model. Let's now look at a linear model using the categorical variable of `Age`. This is the plot of `Age` versus `Weight`:
+
+
+```python
+# HIDDEN
+sns.stripplot(x='Age', y='Weight', data=X_train, order=['<2', '2-5', '5-10', '10-15', '15-20', '>20']);
+```
+
+
+![png](linear_case_study_files/linear_case_study_53_0.png)
+
+
+Seeing how `Age` is a categorical variable, we need to introduce dummy variables in order to produce a linear regression model.
+
+
+```python
+# HIDDEN
+just_age_and_weight = X_train[['Age', 'Weight']]
+with_age_dummies = pd.get_dummies(just_age_and_weight, columns=['Age'])
+model = LinearRegression()
+model.fit(with_age_dummies.drop('Weight', axis=1), with_age_dummies['Weight'])
+
+just_age_and_weight_test = X_test[['Age']]
+with_age_dummies_test = pd.get_dummies(just_age_and_weight_test, columns=['Age'])
+predictions = model.predict(with_age_dummies_test)
+print("MSE:", mse_test_set(predictions))
+```
+
+    MSE: 41511.58282277702
+    
+
+A MSE of around 40000 is worse than what we could get using any single one of the quantitative variables, but this variable could still prove to be useful in our linear model.
+
+Let's try to interpret this linear model. Note that every donkey that falls into an age category, say 2-5 years of age, will receive the same prediction because they all have input data which is a 1 in the 2-5 column and 0 in the rest of the age columns. Thus, we can interpret categorical variables as simply changing the constant in the model because the categorical variable separates the donkeys into groups and gives one prediction for all donkeys within that group.
+
+Our next step is to create a final model using both our categorical variables and multiple quantitative variables.
+
+## Transforming Variables
+
+Recall from our boxplots that `Sex` was not a useful variable, so we will drop it. We will also remove the `WeightAlt` column because we only have its value for 31 donkeys. Also, using `get_dummies`, we transform the categorical variables `BCS` and `Age` into dummy variables so that we can include them in the model. 
+
+
+```python
+# HIDDEN
+X_train = X_train.drop('Weight', axis=1)
+```
+
+
+```python
+donkeys_train = X_train.drop(['Sex', 'WeightAlt'], axis=1)
+donkeys_train = pd.get_dummies(donkeys_train, columns=['BCS', 'Age'])
+donkeys_train.head()
+```
+
+
+
+
```

Comment:
try something like `pandas.set_option('max_columns',15)` so we can see more columns

Reply:
great tip! the notebook looks really cool now :)